src/console_cmds.cpp

```diff
@@ -919,6 +919,12 @@ DEF_CONSOLE_CMD(ConNetworkConnect)
 	IConsolePrintF(CC_DEFAULT, "Connecting to %s...", ip);
 	if (company != nullptr) {
 		join_as = (CompanyID)atoi(company);
+		IConsolePrintF(CC_DEFAULT, "Connecting as company id %d ...", join_as);
+		/* Check if we have a valid company id! */
+		if(!Company::IsValidID(join_as) && join_as != COMPANY_SPECTATOR) {
```

Comment:
This validation is broken in a subtle manner due to line 934 below (join_as--), which has all to do with the showing the users number 1 to 15 whereas the IDs internally are 0 to 14 (with 15 being INVALID_COMPANY).
connect localhost#0 will make you join as spectator (join_as is a byte, so 0 - 1 = 255 which is equal to COMPANY_SPECTATOR).
connect localhost#15 will give you an error message instead of joining company 15
Coding style wise there should be a space between the 'if' and '(', i.e. 'if (' instead of 'if('.